fix: Progress Rehaul (#1443)
appsmithorg/appsmith-docs: website/docs/reference/widgets/progress.md

---
description: Learn how to use the Progress widget to visually track the progress of tasks or processes in your application.
---
# Progress

> This page provides information on using the Progress, which allows users to visualize the progress of certain user-performed or system-triggered actions.

Reviewer: ```suggestion
This page provides information on using the Progress widget, which is used to visualize the progression of specific operations and user or system-triggered actions.

```

<VideoEmbed host="youtube" videoId="Yg1Pfy7uc1s" title="How to use Progress Widget" caption="How to use Progress Widget"/>

## Content properties

These properties are customizable options present in the property pane of the widget, allowing users to modify the widget according to their preferences.

### Basic

#### Infinite loading `boolean`

<dd>

> Enables the widget to enter an infinite loading state, which is beneficial when the progress values cannot be determined. This feature can be activated for queries or API calls that require time to retrieve data.

Reviewer: ```suggestion
Enables the widget to enter an infinite loading state, which is beneficial when the progress values cannot be determined.For instance, this property can be enabled for queries that take a longer time to execute.
```

</dd>

#### Type `string`

<dd>

> Allows you to select the desired format for the progress bar, providing options for either a linear or circular representation of progress.

Reviewer: Options have to be specified separately as in the usual reference format

</dd>

#### Progress `number`

<dd>

> Specify the percentage of progress to be displayed to the user, which can be dynamically updated using values retrieved from queries or JavaScript functions.

Reviewer: ```suggestion
Specify the value of the progress indicator (in percentage). You can also use values retrieved from queries or JavaScript functions within the mustache operator `{{}}`.
```

Reviewer: give an example for the JS part

</dd>

### General

#### Number of steps `number`

<dd>

> Specify the number of steps to break down the progress bar, dividing it into multiple parts with fixed percentages of progress. This configuration allows for better communication and supports only positive integers.

Reviewer: ```suggestion
Specify the number of steps to break down the progress bar into multiple parts with fixed progress percentages. This property only supports positive integers.
```

</dd>

#### Visible `boolean`

<dd>

Controls the visibility of the widget. If you turn off this property, the widget would not be visible in View Mode. Additionally, you can use JavaScript by clicking on **JS** next to the **Visible** property to conditionally control the widget's visibility.

For example, if you want to make the widget visible only when the user selects "Yes" from a Select widget, you can use the following JavaScript expression: 
```js
{{Select1.selectedOptionValue === "Yes"}}
```

</dd>

#### Show result `boolean`

<dd>

> Allows you to enable or disable the display of the current evaluated percentage as a number along with the progress in the widget.

Reviewer: ```suggestion
Control the display of the evaluated percentage as a number along with the progress in the widget.
```

> 

Reviewer: You missed the **Counterclockwise** property for the Circular type progress bar

Author: updated
